specs/WebMessageObjects.md

```diff
@@ -119,6 +138,58 @@ void WebView_WebMessageReceivedHandler(object sender, CoreWebView2WebMessageRece
 }
 ```
 
+## Post a FileSystemHandle to the web content
+Use this API to create a FileSystemHandle from the app and send it to the web content in WebView2.
+
+### Web content code
+```javascript
+chrome.webview.addEventListener("message", function (e) {
+    var fileHandle = e.additionalObjects[0];
+    if (fileHandle.kind === "file") {
+        // run file code
+    } else if (fileHandle.kind === "directory") {
+        // run directory code
+    }
+});
+```
+
+### App code
+```cpp
+wil::com_ptr<ICoreWebView2_21> webview21 =
+    m_webView2.try_query<ICoreWebView2_21>();
+wil::com_ptr<ICoreWebView2Environment> environment =
+    appWindow->GetWebViewEnvironment();
+wil::com_ptr<ICoreWebView2Environment14> environment14 = environment.try_query<ICoreWebView2Environment14>();
+wil::com_ptr<ICoreWebView2FileSystemHandle> fileHandle;
+environment14->CreateWebFileSystemHandle(
+    L"C:\\Users\\<user>\\Documents\\file.txt",
+    COREWEBVIEW2_FILE_SYSTEM_HANDLE_PERMISSION_READWRITE,
+    COREWEBVIEW2_FILE_SYSTEM_HANDLE_KIND_FILE, &fileHandle);
+wil::com_ptr<ICoreWebView2WebObjectCollection> webObjectCollection;
+IUnknown* webObjects[1] = {fileHandle.get()};
+environment14->CreateWebObjectCollection(1, webObjects, &webObjectCollection);

```

Comment:
Would be nice for people copy/pasting and then changing the size of webObjects to use the size of webObjects rather than hardcoding 1 in the call:

```suggestion
environment14->CreateWebObjectCollection(ARRAYSIZE(webObjects), webObjects, &webObjectCollection);

```